### PR TITLE
Add husky, commitlint and eslint

### LINKED ISSUE #27: A husky to run git hooks easily
### Summary

Git hooks are pretty useful. We should look into [husky](https://github.com/typicode/husky).

### Intended Outcome

We should be able to configure hooks for the following tasks:

- Linting commit messages (https://commitlint.js.org/#/).
- Linting code.
- Run tests.

### How will it work?

Follow this [documentation](https://typicode.github.io/husky/#/?id=automatic-recommended).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by alvgaona**
Closes #27

- Adding husky pre-commit, with commitlint, lint, and formatter.
- Update prettier rules and add prettier ignore.
- Comply with format and lint rules.
- Updated `PartialAPIApplicationCommand` to have `name` as required.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-export type PermissionFlags = keyof typeof PermissionFlagsBits;
+export * from 'discord-api-types/v10'
+export * from 'discord-api-types/payloads'
+export type PermissionFlags = keyof typeof PermissionFlagsBits
 
-import type { APIApplicationCommand } from "discord-api-types/payloads";
-export type PartialAPIApplicationCommand = Partial<APIApplicationCommand>;
+type PartialWithRequired<T, K extends keyof T> = Pick<T, K> & Partial<T>
+
+export type PartialAPIApplicationCommand = PartialWithRequired<APIApplicationCommand, 'name'>
```
Comment (alvgaona) [inline src/types.ts:10]: To avoid doing non-null assertions (`!`) in a lot of places.

File: src/handler.ts [hunk 2]
```diff
@@ -1,50 +1,53 @@
...
-import type { CommandInteractionHandler, ComponentInteractionHandler } from "./interaction";
-import type { PartialAPIApplicationCommand } from "./types";
+import type { CommandInteractionHandler, ComponentInteractionHandler } from './interaction'
+import type { PartialAPIApplicationCommand } from './types'
 
-const router = Router();
+const router = Router()
 
-export type Command<DataType extends InteractionDataType | void = void> = [
+export type Command<DataType extends InteractionDataType> = [
```
Comment (alvgaona) [inline src/handler.ts:12]: Using `void` in union types is not a good practice. I don't think we need it anyways.

**Review by alvgaona — COMMENTED** [review]
Context comments.

File: src/types.ts (hunk 1 above)
Comment (Inrixia) [reply src/types.ts:10]: Yea I can see why this is useful

File: src/handler.ts (hunk 2 above)
Comment (alvgaona) [reply src/handler.ts:12]: Yes, I understand. I'd like to find another way to do this without using `void`. The good practices say it's not suitable for types in general. It's just a recommendation anyways. 

> Is there a reason we would want to require people to specify an InteractionDataType every time?

In the Discord API, the application command should always have a type (`CHAT_INPUT`, `USER`, `MESSAGE`). However, it's optional as it defaults to `CHAT_INPUT`. But in our case, it's much more readable to require an explicit type for clarity.

Comment (Inrixia) [reply src/handler.ts:12]: Hmm, we could just use a default other than void, but I think there is a reason I did not do that.

Having it required is fine but does add complexity for beginners

Comment (alvgaona) [reply src/handler.ts:12]: Let me try using a default value.

**Comment by alvgaona** [comment]
@Inrixia I thought it better and I think your suggestion to keep the real history makes much more sense to me. We'll merge commit, not squash and merge. Also, `commitlint` helps us to have a cleaner history, I believe.

File: src/handler.ts (hunk 2 above)
Comment (alvgaona) [reply src/handler.ts]: @Inrixia default value works like a charm.

**Review by gzalo — APPROVED** [review]

**Pull request merged**